packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -801,31 +806,37 @@ abstract class State<T extends StatefulWidget> {
     assert(_debugLifecycleState == _StateLifecycle.created);
   }
 
-  /// Called whenever the configuration changes.
+  /// Called whenever the widget configuration changes.
   ///
   /// If the parent widget rebuilds and request that this location in the tree
   /// update to display a new widget with the same [runtimeType] and [key], the
-  /// framework will update the [config] property of this [State] object to
+  /// framework will update the [widget] property of this [State] object to
   /// refer to the new widget and then call the this method with the previous
   /// widget as an argument.
   ///
-  /// Override this method to respond to changes in the [config] widget (e.g.,
+  /// Override this method to respond to changes in the [widget] widget (e.g.,
   /// to start implicit animations).
   ///
-  /// The framework always calls [build] after calling [didUpdateConfig], which
-  /// means any calls to [setState] in [didUpdateConfig] are redundant.
+  /// The framework always calls [build] after calling [didUpdateWidget], which
+  /// means any calls to [setState] in [didUpdateWidget] are redundant.
   ///
   /// If a [State]'s [build] method depends on an object that can itself change
   /// state, for example a [ChangeNotifier] or [Stream], or some other object to
   /// which one can subscribe to receive notifications, then the [State] should
   /// subscribe to that object during [initState], unsubscribe from the old
   /// object and subscribe to the new object when it changes in
-  /// [didUpdateConfig], and then unsubscribe from the object in [dispose].
+  /// [didUpdateWidget], and then unsubscribe from the object in [dispose].
   ///
   /// If you override this, make sure your method starts with a call to
-  /// super.didUpdateConfig(oldConfig).
+  /// super.didUpdateWidget(oldWidget).
   // TODO(abarth): Add @mustCallSuper.
   @protected
+  void didUpdateWidget(covariant T oldWidget) { }
+
+  /// Deprecated callback when the widget configuration changes.
+  ///
+  /// Use [didUpdateWidget] instead. This method will be removed on 2017/04/20.
+  @Deprecated("Will be removed 2017/04/20")
```

Comment:
ditto here regarding the change suggested for @Deprecated -- either this should be a better message or we should just remove it entirely.